ethclient/ethclient.go

```diff
@@ -696,6 +697,38 @@ func (ec *Client) SendTransaction(ctx context.Context, tx *types.Transaction) er
 	return ec.c.CallContext(ctx, nil, "eth_sendRawTransaction", hexutil.Encode(data))
 }
 
+// SendTransactionSync submits a signed tx and waits for a receipt (or until
+// the optional timeout elapses on the server side). If timeout == 0, the server
+// uses its default.
+func (ec *Client) SendTransactionSync(
+	ctx context.Context,
+	tx *types.Transaction,
+	timeout ...time.Duration,
```

Comment:
that means users can pass in multiple timeout parameters :) I'd make it a pointer and not pass it to the rpc call if it's nil. Or define a new method SendTransactionSyncWithTimeout and SendRawTransactionSyncWithTimeout

Reply:
went with the pointer [b05fbc1](https://github.com/ethereum/go-ethereum/pull/32830/commits/b05fbc18689c54f15bc372c28ab200264f02b421)